### PR TITLE
Http-content-check with new pkg nodeCheck

### PR DESCRIPTION
**Pull request by jdowni000**
Integrated the new pkg nodeCheck into the khcheck http-content-check.

### CODE REVIEW

File: cmd/http-content-check/Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
+  
 build:
-	docker build -t kuberhealthy/http-content-check:v1.3.2 -f Dockerfile ../../
+	docker build -t kuberhealthy/http-content-check:v1.4.2 -f Dockerfile ../../
```
Comment (joshulyne) [inline cmd/http-content-check/Makefile:3]: maybe 1.4.0 instead of 1.4.2?

File: cmd/http-content-check/main.go [hunk 2]
```diff
@@ -42,26 +44,53 @@ func main() {
 	if TargetString == "" {
 		reportErrorAndStop("No string provided in YAML")
 	}
+}
+
+func main() {
+
+	checkTimeLimit := time.Minute * 1
+	ctx, ctxCancel := context.WithTimeout(context.Background(), checkTimeLimit)
```
Comment (joshulyne) [inline cmd/http-content-check/main.go:52]: idt we need ctxCancel -- the context.WithTimeout cancels the context when it reaches `checkTimeLimit` already. so maybe `ctx, _ := context.WithTimeout(context.Background(), checkTimeLimit)` instead?

File: cmd/http-content-check/main.go [hunk 3]
```diff
@@ -42,26 +44,53 @@ func main() {
...
+
+	kubernetesClient, err := kubeClient.Create("")
+	if err != nil {
+		log.Errorln("Error creating kubeClient with error" + err.Error())
+	}
+
+	err = nodeCheck.WaitForKuberhealthy(ctx)
+	if err != nil {
+		ctxCancel()
```
Comment (joshulyne) [inline cmd/http-content-check/main.go:61]: we can get rid of this from the suggestion above ^

File: cmd/http-content-check/main.go [hunk 4]
```diff
@@ -42,26 +44,53 @@ func main() {
...
+		log.Errorln("Error waiting for kuberhealthy endpoint to be contactable by checker pod with error:" + err.Error())
+	}
+
+	err = nodeCheck.WaitForKubeProxy(ctx, kubernetesClient, "kuberhealthy", "kube-system")
+	if err != nil {
+		// ctxCancel()
```
Comment (joshulyne) [inline cmd/http-content-check/main.go:67]: remove ctxCancel!

File: cmd/http-content-check/Makefile (hunk 1 above)
Comment (jdowni000) [reply cmd/http-content-check/Makefile:3]: that makes sense I will adjust

File: cmd/http-content-check/main.go (hunk 2 above)
Comment (jdowni000) [reply cmd/http-content-check/main.go:52]: agreed that is better. I have made adjustment

File: cmd/http-content-check/main.go (hunk 3 above)
Comment (jdowni000) [reply cmd/http-content-check/main.go:61]: done

File: cmd/http-content-check/main.go (hunk 4 above)
Comment (jdowni000) [reply cmd/http-content-check/main.go:67]: done

File: cmd/http-content-check/http-content-check.yaml [hunk 5]
```diff
@@ -7,13 +7,13 @@ spec:
...
-      imagePullPolicy: IfNotPresent # uses local image if present
-      name: main
-      env:
-        - name: "TARGET_URL"
-          value: "http://httpbin.org/" # The URL that application will use to look for a specified string
-        - name: "TARGET_STRING"
-          value: "http" # The string that will be used to parse through provided URL
-        - name: "TIMEOUT_DURATION"
-          value: "30s" # Specifies the time limit for requests made by the client to the URL
+      - image: kuberhealthy/http-content-check:v1.4.2 # The image of the check you just pushed
```
Comment (joshulyne) [inline cmd/http-content-check/http-content-check.yaml:10]: 1.4.2 --> 1.4.0

Comment (jdowni000) [reply cmd/http-content-check/http-content-check.yaml:10]: accidently updated http-check yaml instead....I will fix with a push

**Pull request merged**